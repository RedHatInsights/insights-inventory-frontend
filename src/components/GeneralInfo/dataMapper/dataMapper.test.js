import { render } from '@testing-library/react';
import {
  diskMapper,
  enabledHelper,
  generalMapper,
  interfaceMapper,
  productsMapper,
  repositoriesMapper,
  statusHelper,
} from './dataMapper';

Object.keys(statusHelper).map((oneStatus) => {
  it(`should return ${oneStatus}`, () => {
    const view = render(statusHelper[oneStatus]);
    expect(view.asFragment()).toMatchSnapshot();
  });
});

Object.keys(enabledHelper).map((oneStatus) => {
  it(`should return ${oneStatus}`, () => {
    const view = render(enabledHelper[oneStatus]);
    expect(view.asFragment()).toMatchSnapshot();
  });
});

it('diskMapper', () => {
  expect(
    diskMapper([
      {
        device: 'device',
        label: 'label',
        mountpoint: 'mount',
        options: {
          test: 'data',
        },
        mounttype: 'type',
      },
    ])
  ).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Device",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Label",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Mount point",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Type",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "expandable": true,
      "rows": Array [
        Object {
          "cells": Array [
            "device",
            "label",
            "mount",
            "type",
          ],
          "child": <div>
            test=data
          </div>,
          "isOpen": false,
        },
      ],
    }
  `);
});

it('diskMapper with values', () => {
  expect(
    diskMapper([
      {
        device: { value: 'device' },
        label: { value: 'label' },
        mountpoint: { value: 'mount' },
        options: {
          options: {
            value: { test: 'data' },
          },
        },
        mounttype: 'type',
      },
    ])
  ).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Device",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Label",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Mount point",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Type",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "expandable": true,
      "rows": Array [
        Object {
          "cells": Array [
            "device",
            Object {
              "value": "label",
            },
            "mount",
            "type",
          ],
          "child": <div>
            test=data
          </div>,
          "isOpen": false,
        },
      ],
    }
  `);
});

it('diskMapper - no data', () => {
  expect(diskMapper()).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Device",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Label",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Mount point",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Type",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "expandable": true,
      "rows": Array [],
    }
  `);
});

it('generalMapper', () => {
  expect(generalMapper(['one', 'two'], 'test')).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "test",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "filters": Array [
        Object {
          "type": "textual",
        },
      ],
      "rows": Array [
        Array [
          "one",
        ],
        Array [
          "two",
        ],
      ],
    }
  `);
});

it('generalMapper - no data', () => {
  expect(generalMapper()).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "filters": Array [
        Object {
          "type": "textual",
        },
      ],
      "rows": Array [],
    }
  `);
});

it('interfaceMapper', () => {
  expect(
    interfaceMapper([
      {
        mac_address: 'test-mac',
        mtu: 'test-mtu',
        name: 'test-name',
        state: 'UP',
        type: 'test-type',
      },
      {
        mac_address: 'test-mac2',
        mtu: 'test-mtu2',
        name: 'test-name2',
        state: 'DOWN',
        type: 'test-type2',
      },
      {
        mac_address: 'test-mac2',
        mtu: 'test-mtu2',
        name: 'test-name2',
        state: 'WRONG',
        type: 'test-type2',
      },
    ])
  ).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "MAC address",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "MTU",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        "State",
        Object {
          "title": "Type",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "rows": Array [
        Array [
          "test-mac",
          "test-mtu",
          "test-name",
          Object {
            "title": <Tooltip
              content="Service is running"
            >
              <OutlinedArrowAltCircleUpIcon
                className="ins-c-inventory__detail--up"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          "test-type",
        ],
        Array [
          "test-mac2",
          "test-mtu2",
          "test-name2",
          Object {
            "title": <Tooltip
              content="Service has stopped"
            >
              <OutlinedArrowAltCircleDownIcon
                className="ins-c-inventory__detail--down"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          "test-type2",
        ],
        Array [
          "test-mac2",
          "test-mtu2",
          "test-name2",
          Object {
            "title": <Tooltip
              content="Unknown service status"
            >
              <OutlinedQuestionCircleIcon
                className="ins-c-inventory__detail--unknown"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          "test-type2",
        ],
      ],
    }
  `);
});

it('interfaceMapper - no data', () => {
  expect(interfaceMapper()).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "MAC address",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "MTU",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        "State",
        Object {
          "title": "Type",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "rows": Array [],
    }
  `);
});

it('productsMapper', () => {
  expect(
    productsMapper([
      {
        name: 'test-name',
        status: true,
      },
      {
        name: 'test-name',
        status: false,
      },
      {
        name: 'test-name',
      },
    ])
  ).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        "Status",
      ],
      "rows": Array [
        Array [
          "test-name",
          Object {
            "title": <Tooltip
              content="Unknown service status"
            >
              <OutlinedQuestionCircleIcon
                className="ins-c-inventory__detail--unknown"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
        Array [
          "test-name",
          Object {
            "title": <Tooltip
              content="Unknown service status"
            >
              <OutlinedQuestionCircleIcon
                className="ins-c-inventory__detail--unknown"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
        Array [
          "test-name",
          Object {
            "title": <Tooltip
              content="Unknown service status"
            >
              <OutlinedQuestionCircleIcon
                className="ins-c-inventory__detail--unknown"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
      ],
    }
  `);
});

it('productsMapper - no data', () => {
  expect(productsMapper()).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        "Status",
      ],
      "rows": Array [],
    }
  `);
});

it('repositoriesMapper', () => {
  expect(
    repositoriesMapper({
      enabled: [
        {
          base_url: 'test-url',
          name: 'test-name',
          enabled: true,
          gpgcheck: false,
        },
        {
          base_url: 'test-url',
          name: 'test-name',
          enabled: true,
          gpgcheck: true,
        },
      ],

      disabled: [
        {
          base_url: 'test-url',
          name: 'test-name',
          enabled: false,
          gpgcheck: false,
        },
        {
          base_url: 'test-url',
          name: 'test-name',
          gpgcheck: false,
        },
      ],
    })
  ).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Enabled",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "GPG check",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "filters": Array [
        Object {
          "type": "textual",
        },
        Object {
          "options": Array [
            Object {
              "label": "Is enabled",
              "value": "true",
            },
            Object {
              "label": "Not enabled",
              "value": "false",
            },
          ],
          "type": "checkbox",
        },
        Object {
          "options": Array [
            Object {
              "label": "Is enabled",
              "value": "true",
            },
            Object {
              "label": "Not enabled",
              "value": "false",
            },
          ],
          "type": "checkbox",
        },
      ],
      "rows": Array [
        Array [
          Object {
            "sortValue": "test-name",
            "title": "test-name",
          },
          Object {
            "sortValue": "true",
            "title": <Tooltip
              content="Source enabled"
            >
              <CheckCircleIcon
                className="ins-c-inventory__detail--enabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          Object {
            "sortValue": "false",
            "title": <Tooltip
              content="Source disabled"
            >
              <TimesIcon
                className="ins-c-inventory__detail--disabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
        Array [
          Object {
            "sortValue": "test-name",
            "title": "test-name",
          },
          Object {
            "sortValue": "true",
            "title": <Tooltip
              content="Source enabled"
            >
              <CheckCircleIcon
                className="ins-c-inventory__detail--enabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          Object {
            "sortValue": "true",
            "title": <Tooltip
              content="Source enabled"
            >
              <CheckCircleIcon
                className="ins-c-inventory__detail--enabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
        Array [
          Object {
            "sortValue": "test-name",
            "title": "test-name",
          },
          Object {
            "sortValue": "false",
            "title": <Tooltip
              content="Source disabled"
            >
              <TimesIcon
                className="ins-c-inventory__detail--disabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          Object {
            "sortValue": "false",
            "title": <Tooltip
              content="Source disabled"
            >
              <TimesIcon
                className="ins-c-inventory__detail--disabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
        Array [
          Object {
            "sortValue": "test-name",
            "title": "test-name",
          },
          Object {
            "sortValue": "undefined",
            "title": <Tooltip
              content="Source disabled"
            >
              <TimesIcon
                className="ins-c-inventory__detail--disabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
          Object {
            "sortValue": "false",
            "title": <Tooltip
              content="Source disabled"
            >
              <TimesIcon
                className="ins-c-inventory__detail--disabled"
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              />
            </Tooltip>,
          },
        ],
      ],
    }
  `);
});

it('repositoriesMapper - no data', () => {
  expect(repositoriesMapper()).toMatchInlineSnapshot(`
    Object {
      "cells": Array [
        Object {
          "title": "Name",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "Enabled",
          "transforms": Array [
            [Function],
          ],
        },
        Object {
          "title": "GPG check",
          "transforms": Array [
            [Function],
          ],
        },
      ],
      "filters": Array [
        Object {
          "type": "textual",
        },
        Object {
          "options": Array [
            Object {
              "label": "Is enabled",
              "value": "true",
            },
            Object {
              "label": "Not enabled",
              "value": "false",
            },
          ],
          "type": "checkbox",
        },
        Object {
          "options": Array [
            Object {
              "label": "Is enabled",
              "value": "true",
            },
            Object {
              "label": "Not enabled",
              "value": "false",
            },
          ],
          "type": "checkbox",
        },
      ],
      "rows": Array [],
    }
  `);
});
