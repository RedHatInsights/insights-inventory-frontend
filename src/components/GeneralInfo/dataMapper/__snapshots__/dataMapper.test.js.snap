// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return DOWN 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="ins-c-inventory__detail--down"
    fill="currentColor"
    height="1em"
    role="img"
    style="vertical-align: -0.125em;"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm-32-316v116h-67c-10.7 0-16 12.9-8.5 20.5l99 99c4.7 4.7 12.3 4.7 17 0l99-99c7.6-7.6 2.2-20.5-8.5-20.5h-67V140c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12z"
    />
  </svg>
</DocumentFragment>
`;

exports[`should return UP 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="ins-c-inventory__detail--up"
    fill="currentColor"
    height="1em"
    role="img"
    style="vertical-align: -0.125em;"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      d="M256 504c137 0 248-111 248-248S393 8 256 8 8 119 8 256s111 248 248 248zm0-448c110.5 0 200 89.5 200 200s-89.5 200-200 200S56 366.5 56 256 145.5 56 256 56zm20 328h-40c-6.6 0-12-5.4-12-12V256h-67c-10.7 0-16-12.9-8.5-20.5l99-99c4.7-4.7 12.3-4.7 17 0l99 99c7.6 7.6 2.2 20.5-8.5 20.5h-67v116c0 6.6-5.4 12-12 12z"
    />
  </svg>
</DocumentFragment>
`;

exports[`should return false 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="ins-c-inventory__detail--disabled"
    fill="currentColor"
    height="1em"
    role="img"
    style="vertical-align: -0.125em;"
    viewBox="0 0 352 512"
    width="1em"
  >
    <path
      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
    />
  </svg>
</DocumentFragment>
`;

exports[`should return true 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="ins-c-inventory__detail--enabled"
    fill="currentColor"
    height="1em"
    role="img"
    style="vertical-align: -0.125em;"
    viewBox="0 0 512 512"
    width="1em"
  >
    <path
      d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
    />
  </svg>
</DocumentFragment>
`;
